### PR TITLE
Unit tests for the Ads account controller

### PR DESCRIPTION
**Pull request by mikkamp**
### Changes proposed in this Pull Request:

This PR adds unit tests for the Ads account controller. This re-uses the `RESTControllerUnitTest` which from #973.

Like the previous PR we also change the controller to receive it's dependencies directly in the constructor (instead of through the container). This makes it easier to test and pass mocked services.

See comments from https://github.com/woocommerce/google-listings-and-ads/pull/973#issuecomment-907075512
Issue #388 needs to be addressed before we can refine these tests.

### Detailed test instructions:

1. Run the unit tests and confirm that they all pass.

### Changelog entry

* Add - Unit tests for the Ads account controller.

### CODE REVIEW

**Comment by layoutd** [comment]
Some small rebase issues, but I ended up adding three edge cases that weren't tested and cherry-picking the WooCommerce monorepo fix in the test install script.

File: tests/Framework/RESTControllerUnitTest.php [hunk 1]
```diff
@@ -86,6 +86,7 @@ public function test_register_routes() {
 
 	/**
 	 * Validate that the returned API schema matches what is expected.
+
```
Comment (layoutd) [inline tests/Framework/RESTControllerUnitTest.php:89]: Extra new line
```suggestion
```

**Review by layoutd — APPROVED** [review]
Seems like it's working fine. Will probably need to be merged into https://github.com/woocommerce/google-listings-and-ads/pull/973 before that PR is merged or rebased (or there will be a lot of the same conflicts to resolve).

**Comment by mikkamp** [comment]
Closing in favour of #1208
